connectors/rocketmq-replicator/src/main/java/org/apache/rocketmq/replicator/strategy/DivideTaskByTopic.java

```diff
@@ -28,10 +28,12 @@
 
 public class DivideTaskByTopic extends TaskDivideStrategy {
 
-    @Override public List<KeyValue> divide(Map<String, Set<TaskTopicInfo>> topicRouteMap, TaskDivideConfig tdc) {
+    @Override
+    public List<KeyValue> divide(Map<String, Set<TaskTopicInfo>> topicRouteMap, TaskDivideConfig tdc,
+        int maxTasks) {
 
         List<KeyValue> config = new ArrayList<KeyValue>();
-        int parallelism = tdc.getTaskParallelism();
+        int parallelism = Math.min(tdc.getTaskParallelism(), maxTasks);
```

Comment:
1、may be ConfigDefine.CONN_TASK_PARALLELISM  should  deprecate
   this.taskParallelism = config.getInt(ConfigDefine.CONN_TASK_PARALLELISM, 1);
2、if devide by topic
int parallelism = Math.min(topicNum, maxTasks);

Reply:
> ```
>  1、may be ConfigDefine.CONN_TASK_PARALLELISM  should  deprecate
> ```
> 
> this.taskParallelism = config.getInt(ConfigDefine.CONN_TASK_PARALLELISM, 1); 2、if devide by topic int parallelism = Math.min(topicNum, maxTasks);

as you mean 
if divide by queue, int parallelism = Math.min(queueNum, maxTasks) 
if divide by group, int parallelism = Math.min(groupNum, maxTasks) 
Is that right？
then the number of hash nodes is still compared with taskParallelism and maxTasks?

Reply:
> ```
>  1、may be ConfigDefine.CONN_TASK_PARALLELISM  should  deprecate
> ```
> 
> this.taskParallelism = config.getInt(ConfigDefine.CONN_TASK_PARALLELISM, 1); 2、if devide by topic int parallelism = Math.min(topicNum, maxTasks);

done